presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -767,108 +752,141 @@ public static String keyFromPath(Path path)
+                .serviceConfiguration(s3Configuration)
+                .overrideConfiguration(builder -> {
+                    builder.retryPolicy(retryPolicyBuilder -> retryPolicyBuilder
+                                    .numRetries(maxErrorRetries))
+                            .apiCallTimeout(java.time.Duration.ofMillis(socketTimeout.toMillis()))
+                            .apiCallAttemptTimeout(java.time.Duration.ofMillis(connectTimeout.toMillis()))
+                            .putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_PREFIX,
+                                    userAgentPrefix + " " + S3_USER_AGENT_SUFFIX)
+                            .addMetricPublisher(metricCollector);
+
+                    // Handle signer override if specified
+                    String signerType = hadoopConfig.get(S3_SIGNER_TYPE);
+                    if (signerType != null) {
+                        log.debug("Signer type configuration: %s (Note: v2 handles signers differently)", signerType);
+                    }
+                });
+
         boolean regionOrEndpointSet = false;
 
-        // use local region when running inside of EC2
+        // Use local region when running inside of EC2
         if (pinS3ClientToCurrentRegion) {
-            Region region = Regions.getCurrentRegion();
-            if (region != null) {
-                clientBuilder = clientBuilder.withRegion(region.getName());
-                regionOrEndpointSet = true;
+            try {
+                Region region = DefaultAwsRegionProviderChain.builder().build().getRegion();
+                if (region != null) {
+                    clientBuilder.region(region);
+                    regionOrEndpointSet = true;
+                    log.debug("Using region from provider chain: %s", region);
+                }
+            }
+            catch (Exception e) {
+                log.debug("Could not determine current region from provider chain: %s", e.getMessage());
             }
         }
 
-        String endpoint = hadoopConfig.get(S3_ENDPOINT);
         if (endpoint != null) {
-            clientBuilder = clientBuilder.withEndpointConfiguration(new EndpointConfiguration(endpoint, null));
+            clientBuilder.endpointOverride(URI.create(endpoint));
+            log.debug("Using custom endpoint: %s", endpoint);
+            if (!regionOrEndpointSet) {
+                clientBuilder.region(Region.US_EAST_1);
+                log.debug("Setting default region US_EAST_1 for custom endpoint");
+            }
             regionOrEndpointSet = true;
         }
 
         if (isPathStyleAccess) {
-            clientBuilder = clientBuilder.enablePathStyleAccess();
+            clientBuilder.forcePathStyle(true);
+            log.debug("Using path-style access");
         }
 
         if (!regionOrEndpointSet) {
-            clientBuilder = clientBuilder.withRegion(US_EAST_1);
-            clientBuilder.setForceGlobalBucketAccessEnabled(true);
```

Comment:
We need to migrate this, too. 

We also need to enable the respective alternative to `setForceGlobalBucketAccessEnabled`, that is, `.crossRegionAccessEnabled(true)`.